lib/i18n/backend/base.rb

```diff
@@ -216,34 +216,51 @@ def deep_interpolate(locale, data, values = EMPTY_HASH)
           end
         end
 
+        class LocaleDataDecorator < SimpleDelegator
+          def initialize(*)
+            super
+            @names_symbolized = false
+          end
+
+          def mark_keys_as_symbolized!
+            @names_symbolized = true
+            self
+          end
+
+          def keys_symbolized?
+            @names_symbolized
+          end
+        end
+
         # Loads a single translations file by delegating to #load_rb or
         # #load_yml depending on the file extension and directly merges the
         # data to the existing translations. Raises I18n::UnknownFileType
         # for all other file extensions.
         def load_file(filename)
           type = File.extname(filename).tr('.', '').downcase
           raise UnknownFileType.new(type, filename) unless respond_to?(:"load_#{type}", true)
-          data = send(:"load_#{type}", filename)
-          unless data.is_a?(Hash)
+          locale_data = send(:"load_#{type}", filename)
+          unless locale_data.__getobj__.is_a?(Hash)
```

Comment:
> Not a huge fan of using a decorator here

Fair, this was my main apprehension, thanks for the feedback.

> Have you considered returning two values

My only argument for not returning two values was to maintain the existing method signature that a "hash" (in this case a decorator over `Hash`) is returned. I assumed changing this would have more effect on the code / tests, but the scope seems quite small, so I've gone with that.

> could do something like locale_data.keys.first.is_a?(Symbol)

I thought about that heuristic, but my evaluation was that it seems quite brittle. For instance, in tests, the first level of keys is a symbol, but deeper levels are strings. This wouldn't be a good heuristic in those cases. I suppose we could use a sampling approach to increase the probability, but I think the previous solution is better 👍 

Addressed changes in 86f9919

Reply:
> My only argument for not returning two values was to maintain the existing method signature that a "hash"

If upstream is picky about that, we can suggest putting a magic key in the top level hash instead.